guides/release-workflow.md

```diff
@@ -0,0 +1,162 @@
+
+1. A package is developed on the master branch and released from the master branch. Old versions are not maintained and each new version is released from the master branch.
+2. All work happens on the master branch again. Releases are made on release branches. Bug fixes may be applied to release branches, effectively maintaining old versions, and “hot-fix” releases are made from the release branches.
+
+
+## Good Practicies
+
+There are certain activities which are common for all workflows and are simply good practicies when releasing a package.
+
+1. Version the package using semantic versioning.
+   Details can be found in the
+   [versioning guide ](https://github.com/spacetelescope/style-guides/blob/master/guides/software-versioning.md).
+
+   Before a release update the version of the package to the intended version number. After a release is tagged in the repository and before going back to normal development change the version string in setup.py to the next release, followed by dev. For example, if v 2.1.3 is to be released and the next version will be 2.2.0, before a release is made, set the version string in setup.py to
+   version = “2.1.3”. After the release is done, set the version string to version=”2.2.0.dev”.
+
+   [PEP440](https://www.python.org/dev/peps/pep-0440) describes python package versioning conventions.
+
+2. Maintain a change log for the package.
+
+   It is useful to users if the package has a change log. Many packages maintain a file at the top level, called CHANGES.rst. This file should have a section for each released version and possibly subsections for new features, bug fixes and API changes. Examples cna be found in [astropy](https://github.com/astropy/astropy/blob/master/CHANGES.rst) and [jwst](https://github.com/spacetelescope/jwst/blob/master/CHANGES.rst).
+
+3. When preparing a release use branches and pull requests (PR) following the [Workflow for contributing to a git](https://github.com/spacetelescope/style-guides/blob/master/guides/git-workflow.md).
+
+4. Tag the software with a "release tag" in the repository.
+
+   Create an annotated tag ("release tag") in the repository for each release. The tag should be associated with the branch the release is made from.
+   The command to do this is:
+
+   ```
+   git checkout <release_branch>
+   git tag -a "2.1.3" -m "Tagging release v 2.1.3"
+   ```
+
+5. Publishing a Release on Github
+
+   Online, at the repository root, click on “releases” → “Draft a New Release”.
+   Enter the release tag. Describe the release, enter highlights and publish it.
+
+6. Release on PyPi
+
+   It is common that releases of python packages are published on the Python Packaging Index (PyPi) site. This allows users to find them and install them using `pip`. A PyPi account is required.
```

Comment:
Since the software in `spacetelescope` is being centrally managed to an extent by virtue of it being hosted in that organization, it makes sense to also apply the same controls to how the software is packaged. Astroconda has a central point of control, but PyPI does not. A management account added to each PyPI package would allow publication control, documentation updates, etc, in the same (possibly automated) manner.

This also acknowledges that it is common to publish via PyPI, but are there requirements to do so for particular types of projects? What determines which packages get published there vs. Astroconda vs. conda-forge vs. other places?

Reply:
I can imagine how a scientist may have a package solving a specific issue which they want to manage and make available to others. It doesn't have to be in astroconda but having it on PyPi would be useful. So I think in many cases it's up to the author. But I agree that having an STScI account on the PyPi project page in addition to the author's account would be desirable. However, this is probably more appropriate for a policy. i'm not sure it has a place here, though I can mention it . Whose account do we use in this case?

Reply:
There is a PyPI account `stsci_ssb` which has been added as a maintainer to many, but not all spacetelescope packages. I don't know who has the credentials to it, but I suspect @jhunkeler is one of them.

Reply:
the same is true for readthedocs, we've added an stsci maintainer account which should have admin rights on all the stsci owned packages.